### ISSUE TITLE
ANSIBLE_DIRECTORY will always fail to match if playbook is at the top-level

### ISSUE THREAD

**Issue by aaronsky** [github/super-linter#1110]
## Describe the bug

If `ANSIBLE_DIRECTORY` is set to `/` or some other hacky value to force the container to resolve to `/tmp/lint/`, it will always fail to match `$(dirname /tmp/lint/site.yml)` to `/tmp/lint/` since `dirname` returns `/tmp/lint` (no trailing slash)

## Expected behavior

Because a top-level playbook file is fairly common, I should be able to set my `ANSIBLE_DIRECTORY` to something use a repository root path and have my files identified as Ansible-related. 

## Steps to Reproduce

1. Set `ANSIBLE_DIRECTORY` to `/`
1. Run super-linter either locally or in Actions. For example:
```
docker run \
  -e RUN_LOCAL=true \
  -e ANSIBLE_DIRECTORY="/" \
  -v $HOME/dotfiles:/tmp/lint \
  github/super-linter:latest
```
1. See the debug line in the logs:
```
[0m2020-12-30 21:05:56 [0;34m[DEBUG][0m   Checking if /tmp/lint/dotfiles.yml is an Ansible file. Ansible directory: /tmp/lint/...[0m
[0m2020-12-30 21:05:56 [0;34m[DEBUG][0m   /tmp/lint/dotfiles.yml is NOT an Ansible-related file.[0m
```

## Logs

I've attached logs from my most recent run, using the docker command above.

[super-linter.log](https://github.com/github/super-linter/files/5755729/super-linter.log)

## Additional context

I believe the culprit is this line here: https://github.com/github/super-linter/blob/7943ce1f541fc84bcc0789028dc73aee047f72b3/lib/functions/validation.sh#L143

And for good measure, here's my repository: https://github.com/aaronsky/dotfiles. I'm trying to get `ansible-lint` to pick up `dotfiles.yml`, which is my main playbook.

**Comment by ferrarimarco** [comment]
Being `ANSIBLE_DIRECTORY` relative to `GITHUB_WORKSPACE`, what if you set it to `.`?

**Comment by aaronsky** [comment]
Setting it to `"."` without the changes in #1111 will result in the `ANSIBLE_DIRECTORY` evaluating to `/tmp/lint/.`, which will similarly never be equal to `/tmp/lint`

**Comment by ferrarimarco** [comment]
I see your point now. As said in #1111, to set `ANSIBLE_DIRECTORY` as the root, you should set it to `.`, and we should handle that corner case only, at least for how I'm seeing this. I'll comment in the PR directly (thanks for that!).

All this said, the fix for the hardcoded ansible directory in #1111 is still necessary.
